### ISSUE TITLE
[Bug] Person records are duplicated

### ISSUE THREAD

**Issue by johnclary** [cityofaustin/atd-data-tech#16115]
As identified in #4623, there are duplicate person and primary person records in the DB. We need to understand why this happens and fix it.

It seems that there are at least two different cases of duplicate person records in the DB. In one case, records have the same `crash_id`, `unit_nbr`, `prsn_nbr`, and `prsn_type_id` values. As Mateo pointed out [here](https://github.com/cityofaustin/atd-data-tech/issues/4623#issuecomment-742719431), this seems to be the result of including `prsn_occpnt_pos_id` in the record's unique constraint.

Here is a starter query to identify these cases:

```sql
SELECT DISTINCT ON (p1.person_id)
	*
FROM
	atd_txdot_person p1,
	atd_txdot_person p2
WHERE
	p1.crash_id = p2.crash_id
	AND p1.unit_nbr = p2.unit_nbr
	AND p1.prsn_nbr = p2.prsn_nbr
	AND p1.prsn_type_id = p2.prsn_type_id
	AND p1.person_id != p2.person_id;
```

There appears to be another case, which may be related, in which person **and** unit records are duplicated. See crash ID [19113648](https://visionzero.austin.gov/editor/#/crashes/19113648) for example.

This query finds person records that have the same `crash_id`, `prsn_first_name`, and `prsn_last_name`, excluding crashes where the first or last name is `UNKNOWN`. Bear in mind that the name columns are not available on older records. 

```sql
SELECT
    *
FROM (
    SELECT
        unique_names.name_id,
        count(unique_names.name_id) AS name_count
    FROM (
        SELECT
            prsn_first_name || '_' || prsn_last_name || '_' || crash_id::text AS name_id
        FROM
            atd_txdot_person
        WHERE
            prsn_first_name != 'UNKNOWN'
            AND prsn_last_name != 'UNKNOWN'
            AND prsn_last_name IS NOT NULL) AS unique_names
    GROUP BY
        unique_names.name_id) unique_name_count
WHERE
    unique_name_count.name_count > 1
ORDER BY
    name_count DESC;
```

**Comment by frankhereford** [comment]
# Investigate why we have duplicate records for people / primarypeople

## TLDR

The VZ database and CRIS import script are using 2 additional, wrong columns to uniquely identify persons and primary persons.

## Findings

As you considered in the issue itself, the duplication of person and primary person records is the result of our CRIS import script using more fields than approrpiate to uniquely identify any given unique person in the database. 

To uniquely identify a given, discrete human involved in a crash, you need 3 fields to be distinct, as a group: `crash_id`, `unit_nbr`, `prsn_nbr`. The CRIS import script, however, erroniously adds two additional fields: `prsn_type_id`, `prsn_occpnt_pos`.

The CRIS import isn't the root of the issue, however. The import script is in compliance with the DB's current configuration of a unique constraints found in `atd_txdot_person / atd_txdot_person_unique` and `atd_txdot_primaryperson / atd_txdot_primaryperson_unique`. 

The solution may be to:

* ensure the records are unique over `crash_id`, `unit_nbr`, and `prsn_nbr`
* to modify the database constraints to be correct
* update the CRIS ETL's configuration file to the correct columns
* and purge the older record of the pairs of duplicates.

### Misc

You can see some inline comments surrounding the use of the two fields (`prsn_type_id` and `prsn_occpnt_pos_id`) as part of a unique persons' and primary persons' primary key [here](https://github.com/cityofaustin/atd-vz-data/blob/master/atd-etl/cris_import/lib/mappings.py#L78-L94).

In the same file, you can see the part of the CRIS import configuration that defines the combinations of fields which define the primary key [here](https://github.com/cityofaustin/atd-vz-data/blob/master/atd-etl/cris_import/lib/mappings.py#L23-L36). The CRIS ETL configuration is aligning to the DB's unique constraints on the persons and primary persons tables. 

### Supporting Queries

```sql
select p1.primaryperson_id as p1_person_id, p2.primaryperson_id as p2_person_id, 
	p1.last_update as p1_last_update, p2.last_update as p2_last_update, 
	p1.prsn_occpnt_pos_id as p1_prsn_occpnt_pos_id, p2.prsn_occpnt_pos_id as p2_prsn_occpnt_pos_id,
	p1.prsn_type_id as p1_prsn_type_id, p2.prsn_type_id as p2_prsn_type_id, 
	p1.crash_id, p1.unit_nbr, p1.prsn_nbr
from atd_txdot_primaryperson p1
join atd_txdot_primaryperson p2 on (p1.crash_id = p2.crash_id and p1.unit_nbr = p2.unit_nbr and p1.prsn_nbr = p2.prsn_nbr and p1.primaryperson_id != p2.primaryperson_id)
where p1.primaryperson_id > p2.primaryperson_id -- total set 469
--and (p1.prsn_occpnt_pos_id = p2.prsn_occpnt_pos_id or p1.prsn_type_id = p2.prsn_type_id) -- 350 with this clause
--and p1.prsn_occpnt_pos_id = p2.prsn_occpnt_pos_id -- 250
--and p1.prsn_type_id = p2.prsn_type_id -- 100
--and not (p1.prsn_occpnt_pos_id = p2.prsn_occpnt_pos_id or p1.prsn_type_id = p2.prsn_type_id) -- 116
--and ((p1.prsn_type_id is null or p2.prsn_type_id is null) and (p1.prsn_occpnt_pos_id is null or p2.prsn_occpnt_pos_id is null)) -- 3
order by p1.last_update desc, p1.crash_id desc, p1.unit_nbr asc, p1.prsn_nbr asc
```

You can comment in and out the various lines there in the where clause to see how the various duplicate pairs were formed.

**Comment by frankhereford** [comment]
@johnclary I'm not sure how to submit this for review like we might with a PR. If this is what you needed, do you mind closing this? If not, I'll be super happy to continue investigating.

Additionally, there are findings related to the particular crash you pointed out. Do to the nature of the findings, I will report them to you and the team in slack.
